### ISSUE TITLE
Implement completion

### ISSUE THREAD

**Issue by technomancy** [clojure-emacs/cider#5]
We should have everything we need on the Clojure side from the `clojure-complete` library brought in by `lein repl`, but we should check to see if it's loaded first since it's possible folks could be using `nrepl.el` to connect to nrepl servers embedded in other contexts.

Once completion is in place I would feel comfortable recommending nrepl.el over slime for general use. I'd love to see reload-all support, the inspector, and swank.core/break ported too, but I consider those more as nice-to-haves rather than essentials.
